setup/air-gapped/index.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Air-Gapped Deployment
+description: Learn how to set up an air-gapped Coder deployment.
+---
+
+If you need increased security for your Coder deployments, you can set up an
+air-gapped deployment.
+
+To do so, you must:
+
+- Pull all Coder deployment resources into your air-gapped environment
+- Push the images to your Docker registry,
+- Deploy Coder from within your air-gapped environment
+
+> Coder licenses issued as part of the trial program do not support air-gapped
+> deployments.
+
+## Dependencies
+
+Before proceeding, please ensure that you've installed the following software
+dependencies:
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [helm](https://helm.sh/docs/intro/install/)
+
+Next, configure the following items in the same network as the Kubernetes
+cluster that will run Coder (we've provided links to a suggested option for each
+item type, but you're welcome to use the alternatives of your choice):
+
+- [Docker Registry](https://hub.docker.com/_/registry)
+- A [DNS server](https://coredns.io) (or you can use
+  [HostAliases](https://kubernetes.io/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases/))
+- A
+  [certificate authority](https://github.com/activecm/docker-ca/blob/master/Dockerfile)
+  or [self-signed certificates](#self-signed-certificate-for-the-registry)
+
+## Network configuration
+
+Coder requires several preliminary steps to be performed on your network before
+you can deploy Coder. If don't already have the following on your network,
+please see our [infrastructure setup guide](infrastructure.md):
+
+- A certificate authority
+- A domain name service
+- A local Docker Registry
+
+## Version controlling your changes to the Coder install files
+
+Throughout this article, we will suggest changes to the Helm chart that you'll
+obtain from the `.tgz` that's returned when you run `helm pull`. We recommend
+version controlling your files.
+
+## Step 1: Pull all Coder resources into your air-gapped environment
+
+Coder is deployed through [helm](https://helm.sh/docs/intro/install/), and the
+platform images are hosted in Coder's Docker Hub repo.
+
+1. Pull down the Coder helm charts by running the following in a non-air-gapped
+   environment:
+
+   ```console
+   helm repo add coder https://helm.coder.com
+   helm pull coder/coder
+   ```
+
+   These commands will add Coder's helm charts and pull the latest stable
+   release into a tarball file whose name uses the following format:
+   `coder-X.Y.Z.tgz` (X.Y.Z is the Coder release number).
+
+1. Pull the images for the Coder platform from the following Docker Hub
+   locations:
+
+   [coder-service](https://hub.docker.com/r/coderenvs/coder-service)
+
+   [envbox](https://hub.docker.com/r/coderenvs/envbox)
+
+   [envbuilder](https://hub.docker.com/r/coderenvs/envbuilder)
+
+   [timescale](https://hub.docker.com/r/coderenvs/timescale)
```

Comment:
We really only use timescale for evaluation purposes, and in our helm chart, we recommend that customers bring in their own Postgres, and keep it external to the cluster. 

I think we should clarify this up-front, rather than instructing customers to pull in the timescale image (it also has a bevy of critical vulnerabilities).